// Copyright 2024 InfraMatrix

// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package hdp;

service vmm {
  rpc GetVMS(GetVMSRequest) returns (GetVMSResponse);
  rpc CreateVM(CreateVMRequest) returns (CreateVMResponse);
  rpc DeleteVM(DeleteVMRequest) returns (DeleteVMResponse);
  rpc StartVM(StartVMRequest) returns (StartVMResponse);
  rpc ShutdownVM(ShutdownVMRequest) returns (ShutdownVMResponse);
  rpc ResumeVM(ResumeVMRequest) returns (ResumeVMResponse);
  rpc StopVM(StopVMRequest) returns (StopVMResponse);
  rpc GetVMStatus(GetVMStatusRequest) returns (GetVMStatusResponse);

}

message GetVMSRequest {
  int64 status = 1;
}
message GetVMSResponse {
    repeated string vm_names = 1;
}

message CreateVMRequest {}
message CreateVMResponse {
  string vm_name = 1;
}

message DeleteVMRequest {
  int64 vm_number = 1;
}
message DeleteVMResponse {
  string vm_name = 1;
}

message StartVMRequest {
  int64 vm_number = 1;
}
message StartVMResponse {
  string vm_name = 1;
}

message ShutdownVMRequest {
  int64 vm_number = 1;
}
message ShutdownVMResponse {
  string vm_name = 1;
}

message ResumeVMRequest {
  int64 vm_number = 1;
}
message ResumeVMResponse {
  string vm_name = 1;
}

message StopVMRequest {
  int64 vm_number = 1;
}
message StopVMResponse {
  string vm_name = 1;
}

message GetVMStatusRequest {
  int64 vm_number = 1;
}
message GetVMStatusResponse {
  string vm_status = 1;
}
